### PR TITLE
Example for hough_circle

### PR DESCRIPTION
**Pull request by juliusbierk**

### CODE REVIEW

File: skimage/transform/hough_transform.py [hunk 1]
```diff
@@ -148,6 +148,27 @@ def hough_circle(image, radius, normalize=True, full_output=False):
         Hough transform accumulator for each radius.
         R designates the larger radius if full_output is True.
         Otherwise, R = 0.
+
+    Examples
+    --------
+    >>> from skimage.transform import hough_circle
+    >>> img = np.zeros((100,100),dtype=np.bool_)
```
Comment (emmanuelle) [inline skimage/transform/hough_transform.py]: Please use PEP8 https://www.python.org/dev/peps/pep-0008/

Comment (emmanuelle) [inline skimage/transform/hough_transform.py]: There should be a whitespace after a comma. I was telling you that there would be some nitpicking :-).

File: skimage/transform/hough_transform.py [hunk 2]
```diff
@@ -148,6 +148,27 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+    >>> X,Y = np.meshgrid(np.arange(100),np.arange(100))
+    >>> x0,y0,radius = 20,35,23
```
Comment (emmanuelle) [inline skimage/transform/hough_transform.py]: same here and in the other lines, please add whitespaces

File: skimage/transform/hough_transform.py [hunk 3]
```diff
@@ -148,6 +148,27 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+    >>> circle = np.abs((X-x0)**2+(Y-y0)**2-radius**2)<5**2
```
Comment (emmanuelle) [inline skimage/transform/hough_transform.py]: I'm not sure I understand how you define the circle, isn't it ``circle = X-x0)**2+(Y-y0)**2 < radius**2``?

Comment (juliusbierk) [inline skimage/transform/hough_transform.py]: That would give a filled circle, yes? A perfect circle would be (X-x0)**2+(Y-y0)**2 == radius**2, so I've just moved the radius to the other side and given it a finite thickness.

**Comment by jni** [comment]
@juliusbierk @emmanuelle would this example not be better off as a gallery example? It seems a bit long-winded for a docstring.

If we want to keep it a docstring, I would suggest breaking it up with comments describing each section (making the image, using the filter, etc.)

Thoughts?

**Comment by juliusbierk** [comment]
@jni I see your point. I'll make it shorter. It's a bit too to be a gallery example I think, no?

File: skimage/transform/hough_transform.py [hunk 4]
```diff
@@ -148,6 +148,27 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+    >>> # Find position of circle from known radius:
+    >>> res = hough_circle(img,np.array([radius]))[0,:,:]
+    >>> y,x = np.unravel_index(np.argmax(res),res.shape)
+    >>> x,y
+    (20, 35)
+    >>> # Find position and radius by trying a range of radii:
+    >>> radii_range = np.arange(5,50)
+    >>> res = hough_circle(img,radii_range)
+    >>> ridx,y,x = np.unravel_index(np.argmax(res),res.shape)
+    >>> x,y,radii_range[ridx]
```
Comment (emmanuelle) [inline skimage/transform/hough_transform.py]: >>> radii_range[ridx], x, y 

to keep the same order as the line above

File: skimage/transform/hough_transform.py [hunk 5]
```diff
@@ -148,6 +148,27 @@ def hough_circle(image, radius, normalize=True, full_output=False):
...
+    >>> img[circle] = 1
+    >>> # Find position of circle from known radius:
+    >>> res = hough_circle(img,np.array([radius]))[0,:,:]
+    >>> y,x = np.unravel_index(np.argmax(res),res.shape)
+    >>> x,y
+    (20, 35)
+    >>> # Find position and radius by trying a range of radii:
+    >>> radii_range = np.arange(5,50)
+    >>> res = hough_circle(img,radii_range)
+    >>> ridx,y,x = np.unravel_index(np.argmax(res),res.shape)
```
Comment (emmanuelle) [inline skimage/transform/hough_transform.py]: Add a comment to explain that you're looking for the maximum.

**Comment by emmanuelle** [comment]
@juliusbierk thank you, the example looks even better now. You're still missing a few whitespaces :-).

**Comment by emmanuelle** [comment]
Merged, thank you @juliusbierk !

**Pull request merged**